File: pulsar-third/pulsar-chrome-devtools/src/main/java/com/github/kklisura/cdt/protocol/events/network/RequestWillBeSentExtraInfo.java
package com.github.kklisura.cdt.protocol.events.network;

/*-
 * #%L
 * cdt-java-client
 * %%
 * Copyright (C) 2018 - 2019 Kenan Klisura
 * %%
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * #L%
 */

import com.github.kklisura.cdt.protocol.support.annotations.Experimental;
import com.github.kklisura.cdt.protocol.types.network.BlockedCookieWithReason;
import java.util.List;
import java.util.Map;

/**
 * Fired when additional information about a requestWillBeSent event is available from the network
 * stack. Not every requestWillBeSent event will have an additional requestWillBeSentExtraInfo fired
 * for it, and there is no guarantee whether requestWillBeSent or requestWillBeSentExtraInfo will be
 * fired first for the same request.
 */
@Experimental
public class RequestWillBeSentExtraInfo {

  private String requestId;

  private List<BlockedCookieWithReason> blockedCookies;

  private Map<String, Object> headers;

  /** Request identifier. Used to match this information to an existing requestWillBeSent event. */
  public String getRequestId() {
    return requestId;
  }

  /** Request identifier. Used to match this information to an existing requestWillBeSent event. */
  public void setRequestId(String requestId) {
    this.requestId = requestId;
  }

  /**
   * A list of cookies which will not be sent with this request along with corresponding reasons for
   * blocking.
   */
  public List<BlockedCookieWithReason> getBlockedCookies() {
    return blockedCookies;
  }

  /**
   * A list of cookies which will not be sent with this request along with corresponding reasons for
   * blocking.
   */
  public void setBlockedCookies(List<BlockedCookieWithReason> blockedCookies) {
    this.blockedCookies = blockedCookies;
  }

  /** Raw request headers as they will be sent over the wire. */
  public Map<String, Object> getHeaders() {
    return headers;
  }

  /** Raw request headers as they will be sent over the wire. */
  public void setHeaders(Map<String, Object> headers) {
    this.headers = headers;
  }
}

File: pulsar-core/pulsar-tools/pulsar-browser/src/main/kotlin/ai/platon/cdt/kt/protocol/events/network/RequestWillBeSentExtraInfo.kt
@file:Suppress("unused")
package ai.platon.cdt.kt.protocol.events.network

import ai.platon.cdt.kt.protocol.support.annotations.Experimental
import ai.platon.cdt.kt.protocol.support.annotations.Optional
import ai.platon.cdt.kt.protocol.types.network.BlockedCookieWithReason
import ai.platon.cdt.kt.protocol.types.network.ClientSecurityState
import com.fasterxml.jackson.`annotation`.JsonProperty
import kotlin.Any
import kotlin.String
import kotlin.collections.List
import kotlin.collections.Map

/**
 * Fired when additional information about a requestWillBeSent event is available from the
 * network stack. Not every requestWillBeSent event will have an additional
 * requestWillBeSentExtraInfo fired for it, and there is no guarantee whether requestWillBeSent
 * or requestWillBeSentExtraInfo will be fired first for the same request.
 */
@Experimental
data class RequestWillBeSentExtraInfo(
  @param:JsonProperty("requestId")
  val requestId: String,
  @param:JsonProperty("associatedCookies")
  val associatedCookies: List<BlockedCookieWithReason>,
  @param:JsonProperty("headers")
  val headers: Map<String, Any?>,
  @param:JsonProperty("clientSecurityState")
  @param:Optional
  val clientSecurityState: ClientSecurityState? = null,
)
